### PR TITLE
[INLONG-7879][Manager][Sort][Agent] Multi-version code migration

### LINKED ISSUE #7879: [Improve][Manager][Sort][Agent] Multi-version code migration
### Description

1. Migrate the connector to a separate version module.
2. Added sort-flink-dependencies to flink's dependencies base module.
3. Migrating the centrally managed flink version depends on the root POM.
4. Rebase master.

### InLong Component

InLong Manager, InLong Agent, InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by GanfengTan**
### Prepare a Pull Request
1. In the structural diagram after the architecture adjustment, the version that Flink depends on is managed independently, instead of the unified management of the version of the entire project of INLONG.
image

2. Among them, sort depends on the relevant versions of Flink and is unified in sort-flink-dependencies.

3. Version switching SORT FLINK
a. Change sort-flink modules.
b. alter flink version in the pom of inlong root. 
         <flink.version>xxxxxx</flink.version>
        <flink.minor.version>xxxxxx</flink.minor.version>
c. alter sort base class: 
LogDirtySink
FormatUtils
FormatUtilsTest
(MapNullKeyMode -->  JsonFormatOptions.MapNullKeyMode)
4. Please maintain three versions of connector at the same time and submit for Sort Connector.

- Fixes #7879 

### Motivation

Change the sort structure.

### CODE REVIEW

File: inlong-agent/pom.xml [hunk 1]
```diff
@@ -36,4 +36,54 @@
         <module>agent-release</module>
         <module>agent-docker</module>
     </modules>
+
+    <properties>
+        <debezium.version>1.8.0.Final</debezium.version>
```
Comment (dockerzhang) [inline inlong-agent/pom.xml:41]: it's better to add `<debezium.version>1.8.0.Final</debezium.version>` to root pom?

File: inlong-manager/manager-plugins/pom.xml [hunk 2]
```diff
@@ -28,6 +28,11 @@
     <artifactId>manager-plugins</artifactId>
     <name>Apache InLong - Manager Plugins</name>
 
+    <properties>
+        <flink.version>1.13.5</flink.version>
+        <flink.scala.binary.version>2.11</flink.scala.binary.version>
+    </properties>
```
Comment (dockerzhang) [inline inlong-manager/manager-plugins/pom.xml:34]: ditto

File: inlong-sort/pom.xml [hunk 3]
```diff
@@ -35,18 +35,22 @@
...
-        <rat.basedir>${basedir}</rat.basedir>
         <hbase.version>2.2.3</hbase.version>
-        <iceberg.hive.version>2.3.7</iceberg.hive.version>
         <hudi.hive.version>2.3.7</hudi.hive.version>
+        <iceberg.hive.version>2.3.7</iceberg.hive.version>
+        <flink.jackson.version>2.12.1-13.0</flink.jackson.version>
+        <flink.cdc.base.version>2.3.0</flink.cdc.base.version>
+        <log4j.configurationfile>log4j2-test.properties</log4j.configurationfile>
+        <oracle.jdbc.version>19.3.0.0</oracle.jdbc.version>
+        <rat.basedir>${basedir}</rat.basedir>
```
Comment (dockerzhang) [inline inlong-sort/pom.xml:53]: ditto

File: inlong-sort/sort-flink/pom.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    </parent>
+
+    <artifactId>sort-flink</artifactId>
+    <packaging>pom</packaging>
+    <name>Apache InLong - Sort-Flink</name>
+    <modules>
+        <module>base</module>
+        <module>cdc-base</module>
+        <module>sort-flink-v1.13</module>
+        <!--        <module>sort-flink-v1.15</module>-->
```
Comment (EMsnap) [inline inlong-sort/sort-flink/pom.xml:34]: what's the reason for comment this ?

**Comment by EMsnap** [comment]
could you please add description on readme about how to compile sort with different version of flink ?

File: inlong-manager/manager-plugins/pom.xml (hunk 2 above)
Comment (GanfengTan) [reply inlong-manager/manager-plugins/pom.xml:34]: done

File: inlong-agent/pom.xml (hunk 1 above)
Comment (GanfengTan) [reply inlong-agent/pom.xml:41]: The versions of module dependencies are inconsistent, and there is no way to unify them.

File: inlong-sort/sort-flink/pom.xml (hunk 4 above)
Comment (GanfengTan) [reply inlong-sort/sort-flink/pom.xml:34]: Removed and edited

File: inlong-sort/pom.xml (hunk 3 above)
Comment (GanfengTan) [reply inlong-sort/pom.xml:53]: Sort is mainly about the flink connector related management in sort, and other shared configurations have been moved to the root pom.xml.

**Comment by GanfengTan** [comment]
https://github.com/apache/inlong/pull/7923#issuecomment-1522881245
Added to README of sort module.

**Comment by GanfengTan** [comment]
> could you please add description on readme about how to compile sort with different version of flink ?

Added to README of sort module.

File: inlong-sort/sort-flink/sort-flink-v1.13/pom.xml [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+            </dependency>
+            <dependency>
+                <groupId>io.debezium</groupId>
+                <artifactId>debezium-connector-postgres</artifactId>
+                <version>${debezium.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>io.debezium</groupId>
+                <artifactId>debezium-connector-oracle</artifactId>
+                <version>${debezium.version}</version>
```
Comment (gong) [inline inlong-sort/sort-flink/sort-flink-v1.13/pom.xml:226]: debezium version of postgres and oracle is different

File: inlong-sort/sort-flink/sort-flink-v1.13/sort-connectors-v1.13/doris/pom.xml [hunk 6]
```diff
@@ -20,7 +20,7 @@
     <modelVersion>4.0.0</modelVersion>
     <parent>
         <groupId>org.apache.inlong</groupId>
-        <artifactId>sort-connectors</artifactId>
+        <artifactId>sort-connectors-v1.13</artifactId>
         <version>1.7.0-SNAPSHOT</version>
     </parent>
     <artifactId>sort-connector-doris</artifactId>
```
Comment (gong) [inline inlong-sort/sort-flink/sort-flink-v1.13/sort-connectors-v1.13/doris/pom.xml:26]: Maybe all connector artifactId need append flink version

File: inlong-sort/sort-flink/sort-flink-v1.13/pom.xml (hunk 5 above)
Comment (GanfengTan) [reply inlong-sort/sort-flink/sort-flink-v1.13/pom.xml:226]: This is just the global version definition, and the submodule manages the actual version used.

File: inlong-sort/sort-flink/sort-flink-v1.13/sort-connectors-v1.13/doris/pom.xml (hunk 6 above)
Comment (GanfengTan) [reply inlong-sort/sort-flink/sort-flink-v1.13/sort-connectors-v1.13/doris/pom.xml:26]: There is no need to define the version, the module is closed

Comment (gong) [reply inlong-sort/sort-flink/sort-flink-v1.13/sort-connectors-v1.13/doris/pom.xml:26]: > There is no need to define the version, the module is closed

@GanfengTan All connector jar will deploy to maven repository. Connector artifactId can't be distinguished when multiple flink version package and version is same.

**Review by dockerzhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by yunqingmoswu — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Comment by EMsnap** [comment]
Please don't forget to modify the README and pom (add profile for different versions) @GanfengTan

File: inlong-sort/sort-flink/sort-flink-v1.13/sort-connectors-v1.13/doris/pom.xml (hunk 6 above)
Comment (GanfengTan) [reply inlong-sort/sort-flink/sort-flink-v1.13/sort-connectors-v1.13/doris/pom.xml:26]: @gong Thank you for your suggestion. For the stability of the historical version, the 1.13 connector does not add a version identification, and other flink versions will add the corresponding version.

**Review by gong — APPROVED** [review]

**Comment by GanfengTan** [comment]
> Please don't forget to modify the README and pom (add profile for different versions)

The next PR will add relevant information.

**Pull request merged**